src/lib.rs

```diff
@@ -51,15 +51,55 @@ impl Pool {
         pool
     }
 
+    /// Create a new Pool with `size` threads and given thread config.
+    ///
+    /// If `size` is zero, no threads will be spawned. Threads can
+    /// be added later via `expand`.
+    ///
+    /// NOTE: Since Pool can be freely cloned, it does not represent a unique
+    /// handle to the thread pool. As a consequence, the thread pool is not
+    /// automatically shut down; you must explicitly call `Pool::shutdown` to
+    /// shut down the pool.
+    #[inline]
+    pub fn new_with_thread_config(size: usize, thread_config: ThreadConfig) -> Pool {
+        // Create an empty pool with configuration.
+        let pool = Pool::with_thread_config(thread_config);
+
+        // Start the requested number of threads.
+        for _ in 0..size { pool.expand(); }
+
+        pool
+    }
+
+    /// Create an empty Pool, with no threads, with given thread config.
+    ///
+    /// Note that no jobs will run until `expand` is called and
+    /// worker threads are added.
+    #[inline]
+    pub fn with_thread_config(thread_config: ThreadConfig) -> Pool {
```

Comment:
I would just remove this constructor and rename the other one, if you are supplying additional configuration already it's not a stretch to just explicitly say `0` threads.